src/Tokenizer/Analyzer/Analysis/DataProviderAnalysis.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer\Analysis;
+
+final class DataProviderAnalysis
+{
+    /** @var string */
+    private $name;
+
+    /** @var int */
+    private $nameIndex;
+
+    /** @var array<int> */
+    private $usageIndices;
+
+    /**
+     * @param array<int> $usageIndices
+     */
+    public function __construct(string $name, int $nameIndex, array $usageIndices)
+    {
+        $this->name = $name;
+        $this->nameIndex = $nameIndex;
+        $this->usageIndices = $usageIndices;
```

Comment:
I'm not sure:
- it is an internal class, so "use it at your own risk"
- more validation never hurts

Similar arrays of expected type in other analysis classes are not tested, so let's aim for separate PR, so in case He-Who-Does-Not-Want-To-Be-Put-Against-The-Wall would like to discuss it more/rever earlier will be able to do that easier.